rambo/app.py

```diff
@@ -391,109 +368,34 @@ def up(ctx=None, provider=None,  guest_os=None, ram_size=None, drive_size=None,
-        msg = ('Guest OS "%s" is not in the guest OSes whitelist.\n'
-               'Did you have a typo? We\'ll try anyway.\n'
-               'Here is as list of avalible guest OSes:\n\n'
-               % guest_os)
-        for supported_os in SETTINGS['GUEST_OSES']:
-            msg = msg + '%s\n' % supported_os
-        utils.warn(msg)
-
-    ## ram_size and drive_size (coupled)
-    # Cast to strings if they exist so they can stored as env vars.
-    if ram_size:
-        ram_size = str(ram_size)
-    if drive_size:
-        drive_size = str(drive_size)
-
-    if ram_size and not drive_size:
-        try:
-            drive_size = SETTINGS['SIZES'][ram_size]
-        except KeyError: # Doesn't match, but we'll let them try it.
-            drive_size = SETTINGS['DRIVESIZE_DEFAULT']
-    elif drive_size and not ram_size:
-        try:
-            ram_size = list(SETTINGS['SIZES'].keys())[list(SETTINGS['SIZES'].values()).index(drive_size)]
-        except ValueError: # Doesn't match, but we'll let them try it.
-            ram_size = SETTINGS['RAMSIZE_DEFAULT']
-    elif not ram_size and not drive_size:
-        ram_size = SETTINGS['RAMSIZE_DEFAULT']
-        drive_size = SETTINGS['DRIVESIZE_DEFAULT']
-    # else both exist, just try using them
-
-    set_env_var('ramsize', ram_size)
-    set_env_var('drivesize', drive_size)
-
-    ## ram_size
-    if ram_size not in iter(SETTINGS['SIZES']):
-        msg = ('RAM Size "%s" is not in the RAM sizes list.\n'
-               'Did you have a typo? We\'ll try anyway.\n'
-               'Here is as list of avalible RAM sizes:\n\n'
-               % ram_size)
-        for supported_ram_size in iter(SETTINGS['SIZES']):
-            msg = msg + '%s\n' % supported_ram_size
-        utils.warn(msg)
-
-    ## drive_size
-    if drive_size not in iter(SETTINGS['SIZES'].values()):
-        msg = ('DRIVE Size "%s" is not in the DRIVE sizes list.\n'
-               'Did you have a typo? We\'ll try anyway.\n'
-               'Here is as list of avalible DRIVE sizes:\n\n'
-               % drive_size)
-        for supported_drive_size in iter(SETTINGS['SIZES'].values()):
-            msg = msg + '%s\n' % supported_drive_size
-        utils.warn(msg)
-
-    if machine_type:
-        if provider in ('docker', 'lxc', 'virtualbox'):
-            msg = ('You have selected a machine-type, but are not using\n'
-                   'a cloud provider. You selected %s with %s.\n'
-                   % (machine_type, provider))
-            utils.abort(msg)
-        set_env_var('machinetype', machine_type)
+        set_vagrant_vars(params.get(vagrant_cwd), params.get(vagrant_dotfile_path))
 
+    ## Option Handling - These might modify the params dict or set env vars.
+    params['guest_os'] = options.guest_os_option(params['guest_os'])
+    params['machine_type'] = options.machine_type_option(params['machine_type'])
+    params['provider'] = options.provider_option(params['provider'])
```

Comment:
Note the change here. Now rewriting only one key, and only passing in one key to the handler function. This also has the effect of simplifying the handler function a bit since it no longer needs to be dictionary aware.

Reply:
I'm pretty sure I can get behind that